contracts/QuestFactory.sol

```diff
@@ -549,20 +602,33 @@ contract QuestFactory is Initializable, OwnableUpgradeable, AccessControlUpgrade
         ++currentQuest.numberMinted;
         questContract_.singleClaim(msg.sender);
 
-        if(mintFee > 0) processMintFee();
+        if(mintFee > 0) processMintFee(ref_);
 
         emit Quest1155Claimed(msg.sender, currentQuest.questAddress, questId_, questContract_.rewardToken(), questContract_.tokenId());
+
+        if(ref_ != address(0)) {
+            emit QuestClaimedReferred(msg.sender, currentQuest.questAddress, questId_, questContract_.rewardToken(), questContract_.tokenId(), ref_, referralFee);
```

Comment:
In the case that this does get emitted, do we want to emit both `Quest1155Claimed` and `QuestClaimedReferred`?

Reply:
Would we also want to use the same event type for both erc20 and erc1155 quests? Since both get emitted from QuestFactory, might be difficult telling which type of quest this event refers to (though it does have `questId`, but just thinking here)